### PR TITLE
Selenium

### PR DESCRIPTION
**Pull request by coraxster**
add support for selenium(also selenium grid)

### CODE REVIEW

File: src/ProcessManager/SeleniumManager.php [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+declare(strict_types=1);
+
+namespace Panthere\ProcessManager;
+
+use Facebook\WebDriver\Remote\DesiredCapabilities;
+use Facebook\WebDriver\Remote\RemoteWebDriver;
+use Facebook\WebDriver\WebDriver;
+use \Facebook\WebDriver\WebDriverCapabilities;
+
```
Comment (dunglas) [inline src/ProcessManager/SeleniumManager.php]: Extra line to remove

File: src/ProcessManager/SeleniumManager.php [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ */
+
+declare(strict_types=1);
+
+namespace Panthere\ProcessManager;
+
+use Facebook\WebDriver\Remote\DesiredCapabilities;
+use Facebook\WebDriver\Remote\RemoteWebDriver;
+use Facebook\WebDriver\WebDriver;
+use \Facebook\WebDriver\WebDriverCapabilities;
```
Comment (dunglas) [inline src/ProcessManager/SeleniumManager.php]: The initial backslack must also be removed

File: tests/ProcessManager/SeleniumManagerTest.php [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+declare(strict_types=1);
+
+namespace Panthere\Tests\ProcessManager;
+
+use Panthere\ProcessManager\ChromeManager;
+use Panthere\ProcessManager\SeleniumManager;
+use PHPUnit\Framework\TestCase;
+
+/**
+ * @author Kévin Dunglas <dunglas@gmail.com>
```
Comment (dunglas) [inline tests/ProcessManager/SeleniumManagerTest.php]: Should be changed I guess!

**Comment by coraxster** [comment]
ci passed

File: tests/ProcessManager/SeleniumManagerTest.php (hunk 3 above)
Comment (coraxster) [reply tests/ProcessManager/SeleniumManagerTest.php]: I can't :(
https://github.com/dunglas/panthere/blob/master/.php_cs.dist#L6

Comment (dunglas) [reply tests/ProcessManager/SeleniumManagerTest.php]: It defines the copyright, not the author! You can update this line safely, PHP CS Fixer will not complait.

File: src/ProcessManager/SeleniumManager.php [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+use Facebook\WebDriver\WebDriverCapabilities;
+
+final class SeleniumManager implements BrowserManagerInterface
+{
+    private const DEFAULT_HOST = 'http://127.0.0.1:4444/wd/hub';
+
+    private $host;
+    private $capabilities;
+
+    public function __construct(?string $host = null, ?WebDriverCapabilities $capabilities = null)
```
Comment (dunglas) [inline src/ProcessManager/SeleniumManager.php]: Can be `string $host = self::DEFAULT_HOST` IIRC.

File: src/ProcessManager/SeleniumManager.php [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+namespace Panthere\ProcessManager;
+
+use Facebook\WebDriver\Remote\DesiredCapabilities;
+use Facebook\WebDriver\Remote\RemoteWebDriver;
+use Facebook\WebDriver\WebDriver;
+use Facebook\WebDriver\WebDriverCapabilities;
+
+final class SeleniumManager implements BrowserManagerInterface
+{
+    private const DEFAULT_HOST = 'http://127.0.0.1:4444/wd/hub';
```
Comment (dunglas) [inline src/ProcessManager/SeleniumManager.php]: I'm not even sure we need this constant, can be the default value in the constructor directly.

**Comment by coraxster** [comment]
all fixed :)

File: src/ProcessManager/ChromeManager.php [hunk 6]
```diff
@@ -77,7 +77,7 @@ private function findChromeDriverBinary(): string
         }
     }
 
-    private function getDefaultArguments(): array
+    public function getDefaultArguments(): array
```
Comment (dunglas) [inline src/ProcessManager/ChromeManager.php]: Why this change? If it is just for https://github.com/dunglas/panthere/pull/15/files#diff-0b7a157b54171e8b04337166e6da76d3R47, it is not worth it, just copy them.

**Comment by dunglas** [comment]
Thank you very much @coraxster! You you mind updating the readme to explain how to use Selenium?

**Comment by coraxster** [comment]
Thank you for help :)
No problem.
